src/EnergyPlus/UnitarySystem.cc

```diff
@@ -12170,34 +12170,35 @@ namespace UnitarySystems {
-                if (QActual > SuppHeatCoilLoad) {
-                    auto f = [&state, this, FirstHVACIteration, SuppHeatCoilLoad](Real64 const PartLoadFrac) {
-                        Real64 mdot = min(state.dataLoopNodes->Node(this->m_SuppCoilFluidOutletNodeNum).MassFlowRateMaxAvail,
-                                          this->m_MaxSuppCoilFluidFlow * PartLoadFrac);
-                        state.dataLoopNodes->Node(this->m_SuppCoilFluidInletNode).MassFlowRate = mdot;
-                        WaterCoils::SimulateWaterCoilComponents(
-                            state, this->m_SuppHeatCoilName, FirstHVACIteration, this->m_SuppHeatCoilIndex, 0.0, this->m_FanOpMode, PartLoadFrac);
-                        return SuppHeatCoilLoad;
-                    };
-                    int SolFla; // Flag of solver, num iterations if >0, else error index
-                    General::SolveRoot(state, Acc, MaxIte, SolFla, PartLoadFrac, f, 0.0, 1.0);
-                    this->m_SuppHeatPartLoadFrac = PartLoadFrac;
-                } else {
-                    this->m_SuppHeatPartLoadFrac = 1.0;
-                }
+            } else {
+                mdot = 0.0;
+            }
+            state.dataLoopNodes->Node(this->m_SuppCoilFluidInletNode).MassFlowRate = mdot;
+            //     simulate water coil to find operating capacity
+            WaterCoils::SimulateWaterCoilComponents(state,
+                                                    this->m_SuppHeatCoilName,
+                                                    FirstHVACIteration,
+                                                    this->m_SuppHeatCoilIndex,
+                                                    QActual,
+                                                    this->m_FanOpMode,
+                                                    this->m_SuppHeatPartLoadFrac);
+            if (QActual > SuppHeatCoilLoad) {
+                auto f = [&state, this, FirstHVACIteration, SuppHeatCoilLoad](Real64 const PartLoadFrac) {
+                    Real64 mdot = min(state.dataLoopNodes->Node(this->m_SuppCoilFluidOutletNodeNum).MassFlowRateMaxAvail,
+                                      this->m_MaxSuppCoilFluidFlow * PartLoadFrac);
+                    state.dataLoopNodes->Node(this->m_SuppCoilFluidInletNode).MassFlowRate = mdot;
+                    WaterCoils::SimulateWaterCoilComponents(
+                        state, this->m_SuppHeatCoilName, FirstHVACIteration, this->m_SuppHeatCoilIndex, 0.0, this->m_FanOpMode, PartLoadFrac);
+                    return SuppHeatCoilLoad;
+                };
+                int SolFla; // Flag of solver, num iterations if >0, else error index
+                General::SolveRoot(state, Acc, MaxIte, SolFla, PartLoadFrac, f, 0.0, 1.0);
+                this->m_SuppHeatPartLoadFrac = PartLoadFrac;
+            } else {
+                this->m_SuppHeatPartLoadFrac = (SuppHeatCoilLoad > 0.0) ? 1.0 : 0.0;
             }
```

Comment:
And second is if QActual > SuppHeatCoilLoad (12188) then SolveRoot on the result, else (12201) set supp heater PLR to 1, unless supp heater load is 0 in which case set PLR to 0. If mdot water = 0 then QActual should return as 0, so, hopefully, QActaul > SuppHeatCoilLoad should be false in that case. So at the else here, either the load is greater than what the coil can provide or the coil is off because there is no load. I guess I could have also used `(mdot > 0)` here. There is no change to the other section of this code, it just looks that way since I added the else for mdot = 0 (12176 and 12177).